backend/server/routers/followups.py

```diff
@@ -0,0 +1,90 @@
+from typing import Dict
+from fastapi import APIRouter
+from server.database import coursesCOL
+import json 
+
+router = APIRouter(
+    prefix="/followups",
+    tags=["followups"],
+)
+
+def get_next_term(term: str) -> str:
+    # 'S' -> 'T1'
+    # 'T1' -> 'T2'
+    # 'T2' -> 'T3'
+    # 'T3' -> 'S'
+    if term[0] == 'T':
+        if int(term[-1]) < 3:
+            return 'T' + str(int(term[-1]) + 1)
+        return 'S'
+    return 'T1'
+
+@router.post(
+    "/getFollowups/{origin_course}/{origin_term}",
+    responses={
+        200: {
+            "description": "Returns a list of the most popular followup courses",
+            "content": {
+                "application/json": {
+                    "example": {
+                        "originCourse": "COMP1511",
+                        "originTerm": "T2",
+                        "followups": {
+                            "COMP1521": {
+                                "T3": 339,
+                            },
+                            "COMP1531": {
+                                "T3": 200,
+                            },
+                            "COMP2521": {
+                                "T3": 120,
+                            },
+                            "COMP1511": {
+                                "T3": 45,
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+)
+def get_followups(origin_course: str, origin_term: str) -> dict[str, str | dict[str, dict[str, int]]]:
```

Comment:
For complex return types like this, we should probably start using `typeddicts` but, idk if that's smth we wanna start now or, leave it for a later refactor.

@Peedee2002 thoughts?